tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,443 @@ int main(int argc, char **argv)
+            connection->client_protocol_version = S2N_SSLv2;
+            connection->actual_protocol_version = S2N_SSLv2;
+            connection->server_protocol_version = S2N_SSLv2;
+
+            /* Need to set the server config for tls 1.2 ciphers so that we pass auth validation */
+            EXPECT_SUCCESS(s2n_connection_set_config(connection, server_config));
+
+            static struct s2n_cipher_suite *cipher_suites_sslv2[] = {
+                &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       /* 0xC0,0x23 */
+                &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, /* 0xCC,0xA9 */
+                &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   /* 0xCC,0xA8 */
+                &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+            };
+
+            cipher_preferences = (struct s2n_cipher_preferences) {
+                .count = s2n_array_len(cipher_suites_sslv2),
+                .suites = cipher_suites_sslv2,
+                .allow_chacha20_boosting = true
+            };
+
+            
+            /* Server and client have chacha20 boosting enabled. SSLv2 Server negotiates its most preferred
+             * chacha20 cipher suite (which happens to NOT be the client's most preferred)
+             */
+            {
+                uint8_t ecdhe_rsa_chacha20_boosted_wire[] = {
+                    0x00,
+                    TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                    0x00,
+                    TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256,
+                };
+
+                uint8_t count = sizeof(ecdhe_rsa_chacha20_boosted_wire) / S2N_SSLv2_CIPHER_SUITE_LEN;
+
+                if (s2n_chacha20_poly1305.is_available()) {
+                    EXPECT_SUCCESS(s2n_set_cipher_as_sslv2_server(connection, ecdhe_rsa_chacha20_boosted_wire, count));
+                    EXPECT_EQUAL(connection->secure->cipher_suite, &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256);
+                }
+
+                if (!s2n_chacha20_poly1305.is_available()) {
+                    EXPECT_FAILURE_WITH_ERRNO(s2n_set_cipher_as_sslv2_server(connection, ecdhe_rsa_chacha20_boosted_wire, count), S2N_ERR_CIPHER_NOT_SUPPORTED);
+                }
+            }
+        }
+
+        /* Test ChaCha20 boosting behaviour for tls 1.2 connection w/ that supports only rsa certs */ 
+        {
+            static struct s2n_cipher_suite *cipher_suites_with_tls12_chacha20[] = {
+                    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256,       /* 0xC0,0x23 */
+                    &s2n_ecdhe_ecdsa_with_chacha20_poly1305_sha256, /* 0xCC,0xA9 */
+                    &s2n_ecdhe_rsa_with_chacha20_poly1305_sha256,   /* 0xCC,0xA8 */
+                    &s2n_dhe_rsa_with_chacha20_poly1305_sha256,     /* 0xCC,0xAA */
+            };
```

Comment:
Is this test testing chacha20 boosting? The only RSA cipher suites are chacha20 cipher suites, so s2n_ecdhe_rsa_with_chacha20_poly1305_sha256 would be chosen even without chacha20 boosting.

Repeating my comment from above, I think you need to include a check in every test that you get a different result with and without chacha20 boosting.